libnetwork/drivers/bridge/setup_ip_tables_linux.go

```diff
@@ -252,15 +249,37 @@ func setICMP(ipv iptables.IPVersion, bridgeName string, enable bool) error {
 	return appendOrDelChainRule(icmpRule, "ICMP", enable)
 }
 
-// Append to the filter table's DOCKER chain (the default DROP rule must follow
-// per-port ACCEPT rules, which will be inserted at the top of the chain).
-func defaultDrop(ipv iptables.IPVersion, bridgeName string, enable bool) error {
-	dropRule := iptRule{ipv: ipv, table: iptables.Filter, chain: DockerChain, args: []string{
+func (n *bridgeNetwork) setDefaultForwardRule(
+	ipVersion iptables.IPVersion,
+	bridgeName string,
+) error {
+	// Normally, DROP anything that hasn't been ACCEPTed by a per-port/protocol
+	// rule. This prevents direct access to un-mapped ports from remote hosts
+	// that can route directly to the container's address (by setting up a
+	// route via the host's address).
+	action := "DROP"
+	if n.isGwModeUnprotected(ipVersion) {
```

Comment:
Related to my earlier pondering about locking; this takes a lock here (not really important, just .. thinking out loud I guess)